### DISCUSSION TITLE
Take a zoomed photo

### DISCUSSION THREAD

**Discussion by jbabis007** [Apparence-io/CamerAwesome#318] [Q&A]
Hello, I would like to know if it's possible for the captured image to be identical to what we see in the preview. Let me explain. I set the zoom in the camera to the value '1', which looks good in the preview, but unfortunately, the saved image when I take the photo is not zoomed. Thank you in advance for your help.

**Comment by apalala-dev** [comment]
Can you tell use more details about your configuration?
- Your device (and wether it's emulated or not)
- OS version
- camerawesome version

Thank you for reporting this issue!

**Reply by jbabis007** [reply]
Hello and thank you for your help. So, I am using camerawesome: 1.4.0 on an Android 11 tablet made in China. Since my last post, I have changed my tablet because, in reality, taking photos with flash was not working. Today, I am using an HTC A100 tablet with which I can take photos with flash. I hope this information helps you assist me. Thank you again.

**Reply by apalala-dev** [reply]
Hello, did you try with the HTC A100 tablet as well?
I just tried with a Xiaomi Mi 9T and it works as expected.
If you can share your CamerAwesomeBuilder configuration, that might help reproduce the issue

**Reply by jbabis007** [reply]
Yes, I tried with the HTC A100, but there, even in the preview, the zoom has no effect whether I set it to 0 or 1, it's the same! The image in the preview is identical. In summary, with the first tablet, the flash didn't work but the zoom appeared in the preview but not in the captured photo. With the second tablet, the flash works but there's no zoom in both the preview and the captured image.

Here is the code 

import 'dart:io';
import 'dart:ui' as ui;
import 'package:flutter/material.dart';
import 'package:camerawesome/camerawesome_plugin.dart';
import 'package:flutter/services.dart';
import 'package:path_provider/path_provider.dart';
import 'package:pupilledistance/pubelle.dart';

class Camera04 extends StatefulWidget {
  const Camera04({Key? key}) : super(key: key);

  @override
  State<Camera04> createState() => _Camera04State();
}

class _Camera04State extends State<Camera04> {
  clearCacheFolder() async {
    var cacheFolder = await getTemporaryDirectory();
    cacheFolder.delete(recursive: true);
  }

  @override
  void initState() {
    // TODO: implement initState
    clearCacheFolder();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return WillPopScope(
      onWillPop: () async {
        SystemNavigator.pop();
        return false;
      },
      child: Scaffold(
        body: Padding(
          padding: const EdgeInsets.all(20.0),
          child: Column(
            children: [
              Expanded(child: customCamera(context)),
            ],
          ),
        ),
      ),
    );
  }
}

Widget customCamera(BuildContext context) {
  return CameraAwesomeBuilder.custom(
      zoom: 1,
      aspectRatio: CameraAspectRatios.ratio_16_9,
      flashMode: FlashMode.on,
      builder: (cameraState, previewSize, previewRect) {
        return Align(
          alignment: Alignment.bottomCenter,
          child: Padding(
            padding: const EdgeInsets.all(20),
            child: FloatingActionButton(
                child: Icon(Icons.camera),
                onPressed: () {
                  cameraState.when(onPhotoMode: (picState) {
                    //cameraState.sensorConfig.setFlashMode(FlashMode.on);
                    picState.takePhoto().whenComplete(() {
                      Navigator.of(context)
                          .pushNamed('ImageControl', arguments: {
                        'imageFilePath': cameraState.captureState!.filePath,
                        'previewSize': previewSize,
                        'previewRect': previewRect
                      });
                    });
                  });
                }),
          ),
        );
      },
      saveConfig: SaveConfig.photo(pathBuilder: () async {
        initialCaptureMode:
        CaptureMode.photo;
        final Directory extDir = await getTemporaryDirectory();
        final testDir =
            await Directory('${extDir.path}/test').create(recursive: true);
        return '${testDir.path}/${DateTime.now().millisecondsSinceEpoch}.jpg';
      }));
}

**Reply by jbabis007** [reply]
I just tried it on my Samsung S9+ and everything seems to be working correctly!

**Reply by apalala-dev** [reply]
I tried your example as well on my device and it's working fine. Unfortunately, I think it's related to your devices.
Android tablets are rarely very good with cameras.
We rely on CameraX on the Android side, you may try to see if you can make it work with pure native code but we can't do much to help.

**Reply by jbabis007** [reply]
No luck! I will try other approaches, and if I find something, I will share it with you. Thank you for your help.
